velox/connectors/hive/HiveConfig.cpp

```diff
@@ -283,6 +283,12 @@ uint8_t HiveConfig::parquetWriteTimestampUnit(const Config* session) const {
   return unit;
 }
 
+bool HiveConfig::parquetBinaryAsString(const Config* session) const {
```

Comment:
I don't think we need an option for this, can you just check the table schema (`requestedType`) and use `VARCHAR` accordingly?

Reply:
This is a great suggestion, I will try it out.

Reply:
@Yuhta  this suggestion is very great, but i have some thoughts on this.

1. When Velox reads Parquet files, the type-checking logic is as follows:
	a. A schema is generated based on the metadata of the Parquet file. The types in this schema correspond to the data types in Velox.
	b. It checks whether the requested type matches the data type in the schema generated in step 1. If there is a mismatch, an exception is thrown, as described in my issue.
   Velox ensures data type consistency during data reading through this method.
2. the initial schema is generated based on the metadata of the Parquet file and the ParquetReader does not need the requestedType when reading data.
   Due to the type-checking mentioned above, the requestedType is unnecessary in the ParquetReader.

3. Therefore, I believe we should not implement it based on the requestedType approach. Additionally, my way of using options is inspired by the implementation in Spark.

<img width="1499" alt="image" src="https://github.com/facebookincubator/velox/assets/68981916/2f2eea6f-86a2-424d-b90e-90ba53fb0590">

4. we have some customers with different preferences; some want to read binary data as binary, while others prefer to read it as string.

Reply:
`requestedType` is needed because it can be different for `dataType`, like in this case.  And when it is different, we may need to do data migration.  You should allow the case when `requestedType` is `VARCHAR` and `dataType` is `VARBINARY` and do the conversion in column reader.

Reply:
@Yuhta  it turn out that your method is better, thanks for your suggestion. I have committed again, please help me review it when it's convenient for you.